org.eclipse.acute/src/org/eclipse/acute/dotnetnew/DotnetNewWizard.java

```diff
@@ -56,55 +60,84 @@ public boolean performFinish() {
 		}
 
 		if (!template.isEmpty()) {
-			String createCommand = "dotnet new " + template + " -o " + location.toString();
-
-			Runtime runtime = Runtime.getRuntime();
 			try {
-				Process process = runtime.exec(createCommand);
-				if (process.waitFor() != 0) {
-					MessageDialog.openError(getShell(), "Cannot create dotnet template",
-							"The 'dotnet new' command failed.");
-					return false;
-				}
-			} catch (IOException | InterruptedException e) {
+				getContainer().run(true, true, monitor -> {
+					monitor.beginTask("Creating .NET project",0);
+					String createCommand = "dotnet new " + template + " -o " + location.toString();
+					Runtime runtime = Runtime.getRuntime();
+
+					try {
+						Process process = runtime.exec(createCommand);
+						boolean isProcessDone = false;
+						while (!isProcessDone) {
+							if (monitor.isCanceled()) {
+								process.destroyForcibly();
+							}
+							isProcessDone = process.waitFor(100, TimeUnit.MILLISECONDS);
+						}
+						monitor.done();
+						if (process.exitValue() != 0) {
+							MessageDialog.openError(getShell(), "Cannot create dotnet template",
+									"The 'dotnet new' command exited with :" + process.exitValue());
+						} else {
+							createProject(projectName, location);
+						}
+					} catch (IOException | InterruptedException e) {
+						monitor.done();
+						MessageDialog.openError(getShell(), "Cannot create dotnet template",
+								"The 'dotnet new' command failed.");
+					}
+				});
+			} catch (InvocationTargetException | InterruptedException e) {
 				MessageDialog.openError(getShell(), "Cannot create dotnet template",
-						"The 'dotnet new' command failed.");
-				return false;
+						"The 'dotnet new' command failed");
 			}
+		} else {
+			createProject(projectName, location);
```

Comment:
I don't think we should create an empty project in that case. This is related to the question about why checking for `template.isEmpty()`.
I believe we want to create the project in case `dotnet new` has completed properly)